### DISCUSSION TITLE
How to get Elsa 3 dashboard running?

### DISCUSSION THREAD

**Discussion by samadpaydar3** [elsa-workflows/elsa-core#4179] [General]
Hi,

How can I get Elsa 3's workflow designer (i.e. dashboard) up and running?

For **v2,** I used to run the **build-and-run-dashboard-monolith-with-docker.ps1** file which is found in the repo's root

For **v3**, I tried running **build-and-run-all-in-one-web-docker.sh** from the repo's root, but get the following error, like it fails to install nvm. **Does this shell script intended to get the dashboard running?**
(I'm on Windows 10)

![image](https://github.com/elsa-workflows/elsa-core/assets/102711477/4b051a42-ab2c-4986-88d3-a9bbe459c65a)

----------------------------------------------------------

When running the **Elsa.AllInOne.Web** project in visual studio, I get an empty browser page.

![image](https://github.com/elsa-workflows/elsa-core/assets/102711477/5a987002-1549-4b63-8bbd-61373a263f14)

![image](https://github.com/elsa-workflows/elsa-core/assets/102711477/7bff521b-dc3d-4691-8955-59fad80c435c)

**Comment by sfmskywalker** [comment]
Hi @samadpaydar3,

Thank you for providing detailed information about the issue you're facing with Elsa 3's workflow designer.

When running Elsa from the source, it's essential to build the client-side assets. To do this, you should navigate to the `src/modules/Elsa.Workflows.Designer` directory and execute the following commands:

```bash
npm install
npm run build
```

This will ensure that all necessary client-side assets are built and ready for use.

Alternatively, as you mentioned, Elsa 2 had a shell file `build-and-run-dashboard-monolith-with-docker.ps1`. For Elsa 3, you can use the `build-and-run-all-in-one-web-docker.sh` shell script. However, please ensure that all dependencies, including `nvm`, are correctly installed and configured, especially if you're running it on Windows 10.

If you continue to face issues or encounter any other challenges, please let us know. We're here to help!

**Reply by samadpaydar3** [reply]
Thank you @sfmskywalker for your response and help.
